node/node.go

```diff
@@ -97,6 +110,7 @@ func (n *Node) Run(ctx context.Context) (err error) {
 		<-ctx.Done()
 		n.log.Infow("Shutting down Juno...")
 	}()
+	n.http.Run(ctx)
 	return n.synchronizer.Run(ctx)
```

Comment:
I think this should help us manage the http errors

```suggestion
	syncErrChan := make(chan error)
	httpErrChan := make(chan error)

	go func() {
		syncErrChan <- n.synchronizer.Run(ctx)
	}()
	go func() {
		httpErrChan <- n.http.Run(ctx)
	}()

	syncErr := <-syncErrChan
	httpErr := <-httpErrChan

	if syncErr != nil {
		return syncErr
	}
	if httpErr != nil {
		return httpErr
	}
	return err
```

Reply:
We dont want Run to block tho

Reply:
Run should only return an error if the service fails to start, so there is no reason to push them thru a channel because it will return immediately.

Any errors that the service may encounter in runtime should be handled within that service.

Reply:
There is not much the Node can do if http or synchronizer fails anyways